### PR TITLE
`check-format.pl`: improve handling of preprocessor directives like `#define`, etc.

### PR DESCRIPTION
**Pull request by DDvO**
* report `#if` and `#elif` with constant condition
* improve checks on comment opening: `/*`
* extend checking into macro bodies and other preprocessor directives
* make sure handling of indentation etc. in preprocessor directives is independent of normal code
* improve detection/handling of trailing backslash and essentially empty lines
* improve detection of function definition headers
* improve whitespace reporting on `<op>=`, including (in-)equality
* further fixes for whitespace reporting within `for (...)`
* add some tests
* better reflect in tests that whitespace nits in comments are no longer reported
* improve wording of some reports
* refactor order of state variables
* small further improvements

- [x] tests are added or updated

### CODE REVIEW

**Comment by DDvO** [comment]
Ping for approvals.
As mentioned before, deep reviewing is not expected on updates of this tool :)

**Comment by DDvO** [comment]
Can someone please approve?

**Review by t8m — APPROVED** [review]

**Comment by DDvO** [comment]
This is waiting for 2nd approval by @openssl/committers 
As mentioned, no deep reviewing expected :wink:

File: util/check-format-test-negatives.c
Comment (levitte) [inline util/check-format-test-negatives.c:28]: Isn't this comment still format-tagged?

File: util/check-format-test-negatives.c
Comment (levitte) [inline util/check-format-test-negatives.c:37]: These comment openings are new and note quite our coding style.  Have you found them in aged code?

File: util/check-format-test-negatives.c
Comment (DDvO) [reply util/check-format-test-negatives.c:28]: Thanks for your comments.
Here I had made a copy/paste error - fixed.

File: util/check-format-test-negatives.c
Comment (DDvO) [reply util/check-format-test-negatives.c:37]: There are various occurrences of `**` in the beginning of comment lines - see the output of 
```
find . -name '*.[ch]'  | xargs grep "^[/ ]\*\*[/ ]"
```
while the `/!*` are found only in `krb5/` - see the output of 
```
find . -name '*.[ch]'  | xargs grep "^/\*\!"
```
but they are usual also in many other projects like https://github.com/siemens/libsecutils that make use of Doxygen.

If you prefer, I can make the acceptance of these styles optional (off by default).
Or may it make sense to explicitly allow (part or all of) these in the coding style guidelines?
At least, they are not explicitly forbidden in its _Chapter 8: Commenting_.

Comment (levitte) [reply util/check-format-test-negatives.c:37]: I get it that they do exist a little here and there.  As for options, I personally prefer to be flexible by default unless our coding style really forbids something, *but* you could choose to have a "strict" option that's harsher.  Your call.

Comment (DDvO) [reply util/check-format-test-negatives.c:37]: Pleased to hear. 
Then my preference is, for simplicity:
* not to add a "strict comment" option
* not to touch the coding style, which neither encourages nor forbids these special comment styles.

**Review by t8m — APPROVED** [review]

**Comment by DDvO** [comment]
Thanks @t8m for re-approving.
@levitte can you approve as well?

**Review by levitte — APPROVED** [review]

**Comment by DDvO** [comment]
Merged - thanks @t8m and @levitte

**Pull request closed without merging**
